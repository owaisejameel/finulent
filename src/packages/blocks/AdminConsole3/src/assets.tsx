export const imgPasswordVisible = require("../assets/ic_password_visible.png");
export const imgPasswordInVisible = require("../assets/ic_password_invisible.png");

// Customizable Area Start
export const templateimage = require("../assets/templates.png");
export const checklistimage = require("../assets/checklist.png");
export const uploadimage = require("../assets/upload.png");
export const modalCheck = require("../assets/modalcheck.png");
export const tltemplateimage = require("../assets/tl-template.png");
export const tlchecklistimage = require("../assets/tl-checklist.png");
export const logo = require("../assets/drawerlogo2.png");
export const subfolderimage = require("../assets/companylogo4.png")
export const invoiceimage = require("../assets/InvoiceIcon.png");
export const filter = require("../assets/Filter.png")
export const finulentImageInvoice = require("../assets/Bitmap.png");


// Customizable Area End
