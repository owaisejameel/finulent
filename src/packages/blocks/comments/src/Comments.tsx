import React from "react";
// Customizable Area Start
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  ScrollView,
  TouchableWithoutFeedback,
  FlatList,
  Platform,
} from "react-native";
// Customizable Area End

import CommentController, { Props, configJSON } from "./CommentsController";

export default class Comments extends CommentController {
  constructor(props: Props) {
    super(props);
    // Customizable Area Start
    // Customizable Area End
  }

  // Customizable Area Start
  _renderComment = ({ item }: any) => (
    <View>
      <Text>Commet : {this.getCommentText(item.attributes.comment)}</Text>
      <Text>Created: {item.attributes.created_at}</Text>
      <Text>Updaed : {item.attributes.created_at}</Text>
      <Text>By User: {item.attributes.account.email}</Text>
      <Text>Liked : {item.attributes.liked ? "true" : "false"}</Text>
    </View>
  );
  // Customizable Area End

  render() {
    // Customizable Area Start
    return (
      //Merge Engine DefaultContainer
      <ScrollView keyboardShouldPersistTaps="always" style={styles.container}>
        <TouchableWithoutFeedback
          onPress={() => {
            this.hideKeyboard();
          }}
        >
          {/* Customizable Area Start */}
          {/* Merge Engine UI Engine Code */}
          <View>
            {this.isPlatformWeb() ? (
              <Text
                testID="labelTitle" //Merge Engine::From BDS
                style={styles.title} //UI Engine::From Sketch
              >
                {configJSON.labelTitleText}
              </Text> //UI Engine::From Sketch
            ) : null}

            <View>
              <TouchableOpacity
                testID={"btnCommentTxt"}
                style={styles.viewBtn}
                onPress={async () => {
                  this.props.navigation.navigate("CreateComment");
                }}
              >
                <Text style={styles.viewBtnText}>Add Comment</Text>
              </TouchableOpacity>

              {this.state.commentData && this.state.commentData.length > 0 ? (
                <FlatList
                  style={styles.list}
                  data={this.state.commentData}
                  keyExtractor={(item: any) => item.id}
                  ItemSeparatorComponent={() => {
                    return <View style={styles.separator} />;
                  }}
                  renderItem={(item) => this._renderComment(item)}
                />
              ) : null}
            </View>
          </View>
          {/* Merge Engine UI Engine Code */}
          {/* Customizable Area End */}
        </TouchableWithoutFeedback>
      </ScrollView>
      //Merge Engine End DefaultContainer
    );
    // Customizable Area End
  }
}

// Customizable Area Start
const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingVertical: 16,
    paddingHorizontal: 19,
    marginLeft: "auto",
    marginRight: "auto",
    width: Platform.OS === "web" ? "75%" : "100%",
    maxWidth: 650,
    backgroundColor: "#ffffffff",
  },
  title: {
    fontSize: 15,
    textAlign: "left",
    flexWrap: "wrap",
    width: "97%",
  },
  comment: {
    fontSize: 14,
    color: "#252837",
    paddingLeft: 5,
  },
  viewBtn: {
    backgroundColor: "blue",
    paddingVertical: 8,
    borderRadius: 4,
    borderWidth: 1,
    borderColor: "blue",
  },
  viewBtnText: {
    color: "#fff",
    textAlign: "center",
    fontSize: 16,
  },
  list: {
    marginTop: 15,
    marginBottom: 20,
    backgroundColor: "#FFFFFF",
  },
  separator: {
    marginTop: 10,
    marginBottom: 10,
  },
});
// Customizable Area End
