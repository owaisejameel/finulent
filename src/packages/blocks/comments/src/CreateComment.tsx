/* eslint-disable prettier/prettier */
import React from "react";
// Customizable Area Start
import {
  StyleSheet,
  Text,
  View,
  TextInput,
  ScrollView,
  TouchableWithoutFeedback,
  Platform,
  Dimensions,
  Button
} from "react-native";

const Height  = Dimensions.get("window").height;
// Customizable Area End

import CommentController, {
  Props,
  configJSON
} from "./CommentsController";

export default class CreateComment extends CommentController {
  constructor(props: Props) {
    super(props);
    // Customizable Area Start
    // Customizable Area End
  }

  render() {
    // Customizable Area Start
    return (
      //Merge Engine DefaultContainer
      <ScrollView keyboardShouldPersistTaps="always" style={styles.container}>
        <TouchableWithoutFeedback
          onPress={() => {
            this.hideKeyboard();
          }}
        >
          {/* Customizable Area Start */}
          {/* Merge Engine UI Engine Code */}
          <View style={{height: Height - 200}}>
            {this.isPlatformWeb() ? (
              <Text
                testID="labelTitle" //Merge Engine::From BDS
                style={styles.title} //UI Engine::From Sketch
              >
                {configJSON.labelTitleText}
              </Text> //UI Engine::From Sketch
            ) : null}
          <View style={{height: Height - 260}}>

            <TextInput
              style={[styles.input, styles.border]}
              placeholder={'Write a comment...'}
              multiline={true}
              placeholderTextColor= {'#ddd'}
              // value={props.value}
              onChangeText={(text: string) => {this.setComment(text);}}
            />
            <View style={{position: 'absolute', width: '100%', bottom: 0}}>
              <Button onPress={() => {this.createComment();}} title={'Comment'} />
            </View>
          </View>


          </View>
          {/* Merge Engine UI Engine Code */}
          {/* Customizable Area End */}
        </TouchableWithoutFeedback>
      </ScrollView>
      //Merge Engine End DefaultContainer
    );
    // Customizable Area End
  }
}

// Customizable Area Start
const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingVertical: 16,
    paddingHorizontal: 19,
    marginLeft: "auto",
    marginRight: "auto",
    width: Platform.OS === "web" ? "75%" : "100%",
    maxWidth: 650,
    backgroundColor: "#ffffffff"
  },
  title: {
    marginBottom: 32,
    fontSize: 16,
    textAlign: "left",
    marginVertical: 8
  },
  images: {
    width: 90,
    height: 90,
    borderRadius: 10,
     marginBottom: '5%',
  },
  input: {
    width: '100%',
    paddingVertical: 10,
    borderBottomWidth: 1,
    color: '#252837',
    fontSize: 16,
    marginTop: 10
  },
  border: {
    borderBottomColor: '#eee',
  },
});
// Customizable Area End
