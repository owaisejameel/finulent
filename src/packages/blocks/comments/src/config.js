Object.defineProperty(exports, "__esModule", {
  value: true
});

// Customizable Area Start
exports.commentsContentType = "application/json";
exports.commentsApiMethodType = "GET";
exports.commentPOSTAPiMethod = "POST";
exports.exampleApiContentType = "application/json";
exports.textInputPlaceHolder = "Enter Text";
exports.labelTitleText = "Comments";
exports.labelBodyText = "Comments Body";
exports.hintCountryCode = "Select Country";
exports.errorCountryCodeNotSelected = "Please select country code";
exports.commentEndPoint = "/comments/comments"
exports.likeCommentEndPoint = "/comments/like_comment"
// Customizable Area End
