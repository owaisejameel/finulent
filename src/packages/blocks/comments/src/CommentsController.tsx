import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import MessageEnum, {
  getName,
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";

// Customizable Area Start
// Customizable Area End

export const configJSON = require("./config.js");

export interface Props {
  navigation: any;
  id: string;
  // Customizable Area Start
  // Customizable Area End
}
interface S {
  // Customizable Area Start
  commentData: any;
  token: string;
  errorMsg: string;
  loading: boolean;
  comment: string;
  commentsArray: any;
  replyCommentId: any;
  // Customizable Area End
}
interface SS {
  id: any;
}

export default class CommentController extends BlockComponent<Props, S, SS> {
  apiCommentItemCallId: string = "";
  commentApiCallId: string = "";
  createCommentId: string = "";
  likeCommentId: string = "";

  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);
    console.disableYellowBox = true;
    // Customizable Area Start
    this.subScribedMessages = [
      getName(MessageEnum.RestAPIResponceMessage),
      getName(MessageEnum.SessionResponseMessage),
    ];

    this.state = {
      commentData: [],
      errorMsg: "",
      token: "",
      loading: false,
      comment: "",
      commentsArray: [],
      replyCommentId: null,
    };
    // Customizable Area End
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }

  async componentDidMount() {
    super.componentDidMount();
    if (this.isPlatformWeb() === false) {
      this.props.navigation.addListener("willFocus", () => {
        this.getToken();
      });
    } else {
      this.getToken();
    }
    // Customizable Area Start
    // Customizable Area End
  }

  getToken = () => {
    const msg: Message = new Message(
      getName(MessageEnum.SessionRequestMessage)
    );
    this.send(msg);
  };

  async receive(from: string, message: Message) {
    // Customizable Area Start
    runEngine.debugLog("Message Recived", message);
    if (getName(MessageEnum.SessionResponseMessage) === message.id) {
      let token = message.getData(getName(MessageEnum.SessionResponseToken));
      this.setState({ token: token, loading: true }, () => {
        this.getCommentData();
      });
    } else if (getName(MessageEnum.RestAPIResponceMessage) === message.id) {
      const apiRequestCallId = message.getData(
        getName(MessageEnum.RestAPIResponceDataMessage)
      );

      var responseJson = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );

      if (responseJson && !responseJson.errors && responseJson.data) {
        if (responseJson.data.length === 0) {
          this.setState({
            errorMsg: "Data Not Found",
            loading: false,
          });
        } else {
          this.setState({
            commentData: responseJson.data,
            errorMsg: "",
            loading: false,
          });
        }
      } else {
        var errorReponse = message.getData(
          getName(MessageEnum.RestAPIResponceErrorMessage)
        );
        console.log("error resp", errorReponse);
        if (errorReponse === undefined) {
          this.setState({
            errorMsg: "Something went wrong",
            loading: false,
          });
        } else {
          this.setState({
            errorMsg: errorReponse,
            loading: false,
          });
        }
      }
      if (apiRequestCallId === this.createCommentId) {
        if (responseJson != null) {
          if (!responseJson.errors) {
            this.props.navigation.goBack();
          } else {
            //Check Error Response
            this.parseApiErrorResponse(responseJson);
          }
        }
        this.parseApiCatchErrorResponse(errorReponse);
      } else if (apiRequestCallId === this.likeCommentId) {
        if (responseJson != null) {
          if (!responseJson.errors) {
            this.getCommentData();
          } else {
            //Check Error Response
            this.parseApiErrorResponse(responseJson);
          }
        }
        this.parseApiCatchErrorResponse(errorReponse);
      }
    }
    // Customizable Area End
  }

  // Customizable Area Start
  getCommentData(): boolean {
    const header = {
      "Content-Type": configJSON.commentsContentType,
      token: this.state.token,
    };
    console.log("header  ", JSON.stringify(header));
    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );
    console.log("requestMessage, ", requestMessage);

    this.apiCommentItemCallId = requestMessage.messageId;

    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      configJSON.commentEndPoint
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      configJSON.commentsApiMethodType
    );

    runEngine.sendMessage(requestMessage.id, requestMessage);

    return true;
  }

  setComment = (comment: string) => {
    this.setState({ comment });
  };

  createComment = (id?: any, postId?: any) => {
    const header = {
      "Content-Type": configJSON.commentsContentType,
      token: this.state.token,
    };

    const data = {
      attributes: {
        post_id: postId ? postId : 1,
        comment: this.state.comment,
      },
    };
    const httpBody = {
      data: data,
    };
    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );

    this.createCommentId = requestMessage.messageId;
    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      configJSON.commentEndPoint
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestBodyMessage),
      JSON.stringify(httpBody)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      configJSON.commentPOSTAPiMethod
    );
    runEngine.sendMessage(requestMessage.id, requestMessage);
  };

  likeChildComments = (commentId: any) => {
    const header = {
      "Content-Type": configJSON.commentsContentType,
      token: this.state.token,
    };
    const data = {
      attributes: {
        comment_id: commentId,
      },
    };
    const httpBody = {
      data: data,
    };

    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );

    this.likeCommentId = requestMessage.messageId;
    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      configJSON.likeCommentEndPoint
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestBodyMessage),
      JSON.stringify(httpBody)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      configJSON.commentPOSTAPiMethod
    );
    runEngine.sendMessage(requestMessage.id, requestMessage);
  };

  getCommentText(comment: any) {
    try {
      return JSON.parse(comment.replace("=>", ":"))?.text;
    } catch {
      return comment;
    }
  }
  // Customizable Area End
}
