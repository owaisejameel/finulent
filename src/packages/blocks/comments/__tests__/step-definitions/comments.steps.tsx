import { defineFeature, loadFeature} from "jest-cucumber"
import { shallow, ShallowWrapper } from 'enzyme'
import * as helpers from '../../../../framework/src/Helpers'
import {runEngine} from '../../../../framework/src/RunEngine'
import {Message} from "../../../../framework/src/Message"

import MessageEnum, {getName} from "../../../../framework/src/Messages/MessageEnum"; 
import React from "react";
import Comments from "../../src/Comments";
import CreateComment from "../../src/CreateComment";

const navigation = require("react-navigation")

const screenProps = {
    navigation: navigation,
    id: "Comments"
  }

const feature = loadFeature('./__tests__/features/comments-scenario.feature');

defineFeature(feature, (test) => {
    beforeEach(() => {
        jest.resetModules()
        jest.doMock('react-native', () => ({ Platform: { OS: 'web' }}))
        jest.spyOn(helpers, 'getOS').mockImplementation(() => 'web');
    });

    test('User navigates to Comments', ({ given, when, then }) => {
        let commentsBlock: ShallowWrapper;
        let instance: Comments;

        given('I am a User loading Comments', () => {
          commentsBlock = shallow(<Comments {...screenProps}/>)
         });

        when('I navigate to the Comments', () => {
             instance = commentsBlock.instance() as Comments;
        });

        then('Comments will load with out errors', () => {
            
          expect(commentsBlock).toBeTruthy()

          const tokenMsg: Message = new Message(getName(MessageEnum.SessionResponseMessage));
          tokenMsg.addData(getName(MessageEnum.SessionResponseToken), "TOKEN");
          runEngine.sendMessage("Unit Test", tokenMsg);
        
          instance = commentsBlock.instance() as Comments;
          const msgCommentsAPI = new Message(
            getName(MessageEnum.RestAPIResponceMessage)
          );
          msgCommentsAPI.addData(
            getName(MessageEnum.RestAPIResponceDataMessage),
            msgCommentsAPI.messageId
          );
          msgCommentsAPI.addData(
            getName(MessageEnum.RestAPIResponceSuccessMessage),
            {
              "data": [
                  {
                      "id": "1",
                      "type": "comment",
                      "attributes": {
                          "id": 1,
                          "account_id": 1,
                          "post_id": 1,
                          "comment": "{\"text\"=>\"Comment text\"}",
                          "created_at": "2022-01-31T17:36:18.824Z",
                          "updated_at": "2022-01-31T17:36:18.824Z",
                          "post": {
                              "id": 1,
                              "name": "22",
                              "description": "333",
                              "category_id": 6,
                              "created_at": "2022-01-31T14:57:01.045Z",
                              "updated_at": "2022-01-31T14:57:01.045Z",
                              "body": "test body",
                              "location": "test location",
                              "account_id": 1
                          },
                          "account": {
                              "id": 1,
                              "first_name": "test",
                              "last_name": "account",
                              "full_phone_number": "",
                              "country_code": null,
                              "phone_number": null,
                              "email": "tester@me.com",
                              "activated": true,
                              "device_id": null,
                              "unique_auth_id": null,
                              "password_digest": "$2a$12$3fjk1H3xaJi7iT91CrQtmOrrD4QwYdXzIKY/8jWxR58wdt6WykOXW",
                              "created_at": "2022-01-11T15:22:38.266Z",
                              "updated_at": "2022-01-11T15:22:38.266Z",
                              "user_name": null,
                              "role_id": null
                          },
                          "liked": null
                      }
                  },
                  {
                      "id": "2",
                      "type": "comment",
                      "attributes": {
                          "id": 2,
                          "account_id": 1,
                          "post_id": 1,
                          "comment": "{\"text\"=>\"I Love this stuff\"}",
                          "created_at": "2022-01-31T18:36:20.239Z",
                          "updated_at": "2022-01-31T18:36:20.239Z",
                          "post": {
                              "id": 1,
                              "name": "22",
                              "description": "333",
                              "category_id": 6,
                              "created_at": "2022-01-31T14:57:01.045Z",
                              "updated_at": "2022-01-31T14:57:01.045Z",
                              "body": "test body",
                              "location": "test location",
                              "account_id": 1
                          },
                          "account": {
                              "id": 1,
                              "first_name": "test",
                              "last_name": "account",
                              "full_phone_number": "",
                              "country_code": null,
                              "phone_number": null,
                              "email": "tester@me.com",
                              "activated": true,
                              "device_id": null,
                              "unique_auth_id": null,
                              "password_digest": "$2a$12$3fjk1H3xaJi7iT91CrQtmOrrD4QwYdXzIKY/8jWxR58wdt6WykOXW",
                              "created_at": "2022-01-11T15:22:38.266Z",
                              "updated_at": "2022-01-11T15:22:38.266Z",
                              "user_name": null,
                              "role_id": null
                          },
                          "liked": null
                      }
                  },
                  {
                      "id": "3",
                      "type": "comment",
                      "attributes": {
                          "id": 3,
                          "account_id": 1,
                          "post_id": 2,
                          "comment": "{\"text\"=>\"I Love this stuff\"}",
                          "created_at": "2022-01-31T18:37:40.214Z",
                          "updated_at": "2022-01-31T18:37:40.214Z",
                          "post": {
                              "id": 2,
                              "name": "22",
                              "description": "333",
                              "category_id": 6,
                              "created_at": "2022-01-31T18:37:22.252Z",
                              "updated_at": "2022-01-31T18:37:22.252Z",
                              "body": "test body",
                              "location": "test location",
                              "account_id": 1
                          },
                          "account": {
                              "id": 1,
                              "first_name": "test",
                              "last_name": "account",
                              "full_phone_number": "",
                              "country_code": null,
                              "phone_number": null,
                              "email": "tester@me.com",
                              "activated": true,
                              "device_id": null,
                              "unique_auth_id": null,
                              "password_digest": "$2a$12$3fjk1H3xaJi7iT91CrQtmOrrD4QwYdXzIKY/8jWxR58wdt6WykOXW",
                              "created_at": "2022-01-11T15:22:38.266Z",
                              "updated_at": "2022-01-11T15:22:38.266Z",
                              "user_name": null,
                              "role_id": null
                          },
                          "liked": null
                      }
                  },
                  {
                      "id": "4",
                      "type": "comment",
                      "attributes": {
                          "id": 4,
                          "account_id": 1,
                          "post_id": 1,
                          "comment": "{\"text\"=>\"I Love this stuff\"}",
                          "created_at": "2022-01-31T20:11:55.206Z",
                          "updated_at": "2022-01-31T20:11:55.206Z",
                          "post": {
                              "id": 1,
                              "name": "22",
                              "description": "333",
                              "category_id": 6,
                              "created_at": "2022-01-31T14:57:01.045Z",
                              "updated_at": "2022-01-31T14:57:01.045Z",
                              "body": "test body",
                              "location": "test location",
                              "account_id": 1
                          },
                          "account": {
                              "id": 1,
                              "first_name": "test",
                              "last_name": "account",
                              "full_phone_number": "",
                              "country_code": null,
                              "phone_number": null,
                              "email": "tester@me.com",
                              "activated": true,
                              "device_id": null,
                              "unique_auth_id": null,
                              "password_digest": "$2a$12$3fjk1H3xaJi7iT91CrQtmOrrD4QwYdXzIKY/8jWxR58wdt6WykOXW",
                              "created_at": "2022-01-11T15:22:38.266Z",
                              "updated_at": "2022-01-11T15:22:38.266Z",
                              "user_name": null,
                              "role_id": null
                          },
                          "liked": null
                      }
                  },
                  {
                      "id": "5",
                      "type": "comment",
                      "attributes": {
                          "id": 5,
                          "account_id": 1,
                          "post_id": 1,
                          "comment": "333",
                          "created_at": "2022-01-31T20:13:01.807Z",
                          "updated_at": "2022-01-31T20:13:01.807Z",
                          "post": {
                              "id": 1,
                              "name": "22",
                              "description": "333",
                              "category_id": 6,
                              "created_at": "2022-01-31T14:57:01.045Z",
                              "updated_at": "2022-01-31T14:57:01.045Z",
                              "body": "test body",
                              "location": "test location",
                              "account_id": 1
                          },
                          "account": {
                              "id": 1,
                              "first_name": "test",
                              "last_name": "account",
                              "full_phone_number": "",
                              "country_code": null,
                              "phone_number": null,
                              "email": "tester@me.com",
                              "activated": true,
                              "device_id": null,
                              "unique_auth_id": null,
                              "password_digest": "$2a$12$3fjk1H3xaJi7iT91CrQtmOrrD4QwYdXzIKY/8jWxR58wdt6WykOXW",
                              "created_at": "2022-01-11T15:22:38.266Z",
                              "updated_at": "2022-01-11T15:22:38.266Z",
                              "user_name": null,
                              "role_id": null
                          },
                          "liked": null
                      }
                  },
                  {
                      "id": "6",
                      "type": "comment",
                      "attributes": {
                          "id": 6,
                          "account_id": 1,
                          "post_id": 1,
                          "comment": "My comment is the best....\n",
                          "created_at": "2022-01-31T20:17:25.009Z",
                          "updated_at": "2022-01-31T20:17:25.009Z",
                          "post": {
                              "id": 1,
                              "name": "22",
                              "description": "333",
                              "category_id": 6,
                              "created_at": "2022-01-31T14:57:01.045Z",
                              "updated_at": "2022-01-31T14:57:01.045Z",
                              "body": "test body",
                              "location": "test location",
                              "account_id": 1
                          },
                          "account": {
                              "id": 1,
                              "first_name": "test",
                              "last_name": "account",
                              "full_phone_number": "",
                              "country_code": null,
                              "phone_number": null,
                              "email": "tester@me.com",
                              "activated": true,
                              "device_id": null,
                              "unique_auth_id": null,
                              "password_digest": "$2a$12$3fjk1H3xaJi7iT91CrQtmOrrD4QwYdXzIKY/8jWxR58wdt6WykOXW",
                              "created_at": "2022-01-11T15:22:38.266Z",
                              "updated_at": "2022-01-11T15:22:38.266Z",
                              "user_name": null,
                              "role_id": null
                          },
                          "liked": null
                      }
                  },
                  {
                      "id": "7",
                      "type": "comment",
                      "attributes": {
                          "id": 7,
                          "account_id": 1,
                          "post_id": 1,
                          "comment": "hi",
                          "created_at": "2022-01-31T20:31:13.483Z",
                          "updated_at": "2022-01-31T20:31:13.483Z",
                          "post": {
                              "id": 1,
                              "name": "22",
                              "description": "333",
                              "category_id": 6,
                              "created_at": "2022-01-31T14:57:01.045Z",
                              "updated_at": "2022-01-31T14:57:01.045Z",
                              "body": "test body",
                              "location": "test location",
                              "account_id": 1
                          },
                          "account": {
                              "id": 1,
                              "first_name": "test",
                              "last_name": "account",
                              "full_phone_number": "",
                              "country_code": null,
                              "phone_number": null,
                              "email": "tester@me.com",
                              "activated": true,
                              "device_id": null,
                              "unique_auth_id": null,
                              "password_digest": "$2a$12$3fjk1H3xaJi7iT91CrQtmOrrD4QwYdXzIKY/8jWxR58wdt6WykOXW",
                              "created_at": "2022-01-11T15:22:38.266Z",
                              "updated_at": "2022-01-11T15:22:38.266Z",
                              "user_name": null,
                              "role_id": null
                          },
                          "liked": null
                      }
                  },
                  {
                      "id": "8",
                      "type": "comment",
                      "attributes": {
                          "id": 8,
                          "account_id": 1,
                          "post_id": 1,
                          "comment": "Reply to ite\n",
                          "created_at": "2022-01-31T20:31:57.894Z",
                          "updated_at": "2022-01-31T20:31:57.894Z",
                          "post": {
                              "id": 1,
                              "name": "22",
                              "description": "333",
                              "category_id": 6,
                              "created_at": "2022-01-31T14:57:01.045Z",
                              "updated_at": "2022-01-31T14:57:01.045Z",
                              "body": "test body",
                              "location": "test location",
                              "account_id": 1
                          },
                          "account": {
                              "id": 1,
                              "first_name": "test",
                              "last_name": "account",
                              "full_phone_number": "",
                              "country_code": null,
                              "phone_number": null,
                              "email": "tester@me.com",
                              "activated": true,
                              "device_id": null,
                              "unique_auth_id": null,
                              "password_digest": "$2a$12$3fjk1H3xaJi7iT91CrQtmOrrD4QwYdXzIKY/8jWxR58wdt6WykOXW",
                              "created_at": "2022-01-11T15:22:38.266Z",
                              "updated_at": "2022-01-11T15:22:38.266Z",
                              "user_name": null,
                              "role_id": null
                          },
                          "liked": null
                      }
                  },
                  {
                      "id": "9",
                      "type": "comment",
                      "attributes": {
                          "id": 9,
                          "account_id": 1,
                          "post_id": 1,
                          "comment": "ABC",
                          "created_at": "2022-01-31T20:33:49.977Z",
                          "updated_at": "2022-01-31T20:33:49.977Z",
                          "post": {
                              "id": 1,
                              "name": "22",
                              "description": "333",
                              "category_id": 6,
                              "created_at": "2022-01-31T14:57:01.045Z",
                              "updated_at": "2022-01-31T14:57:01.045Z",
                              "body": "test body",
                              "location": "test location",
                              "account_id": 1
                          },
                          "account": {
                              "id": 1,
                              "first_name": "test",
                              "last_name": "account",
                              "full_phone_number": "",
                              "country_code": null,
                              "phone_number": null,
                              "email": "tester@me.com",
                              "activated": true,
                              "device_id": null,
                              "unique_auth_id": null,
                              "password_digest": "$2a$12$3fjk1H3xaJi7iT91CrQtmOrrD4QwYdXzIKY/8jWxR58wdt6WykOXW",
                              "created_at": "2022-01-11T15:22:38.266Z",
                              "updated_at": "2022-01-11T15:22:38.266Z",
                              "user_name": null,
                              "role_id": null
                          },
                          "liked": null
                      }
                  }
              ],
              "meta": {
                  "message": "List of comments created by user."
              }
          }
          );
          instance.commentApiCallId = msgCommentsAPI.messageId;
          runEngine.sendMessage("Unit Test", msgCommentsAPI);

          instance.commentApiCallId = "";
          instance.apiCommentItemCallId = msgCommentsAPI.messageId;
          runEngine.sendMessage("Unit Test", msgCommentsAPI);

          instance.apiCommentItemCallId = "";
          instance.likeCommentId = msgCommentsAPI.messageId;
          runEngine.sendMessage("Unit Test", msgCommentsAPI);

          
          // instance.apiCommentItemCallId = msgValidationAPI.messageId;

          instance.likeChildComments(1);
          instance.getCommentText("{\"text\":\"Message\"}");
          instance.getCommentText("Message");

          expect(commentsBlock).toBeTruthy();

        });

        then('I can leave the screen with out errors', () => {
          let buttonComponent = commentsBlock.findWhere((node) => node.prop('testID') === 'btnCommentTxt');
          buttonComponent.simulate('press')
          instance.componentWillUnmount()
          expect(commentsBlock).toBeTruthy()
        });
    });

    test("User navigates to CreateComment", ({ given, when, then }) => {
      let commentsBlock: ShallowWrapper;
      let instance: CreateComment;
  
      given("I am a User loading CreateComment", () => {
        commentsBlock = shallow(<CreateComment {...screenProps} />);
      });
  
      when("I navigate to the CreateComment", () => {
        instance = commentsBlock.instance() as CreateComment; 
    
      });
  
      then("CreateComment will load with out errors", () => {
        expect(commentsBlock).toBeTruthy();
        expect(commentsBlock).toMatchSnapshot();
        
        const msgValidationAPI = new Message(
          getName(MessageEnum.RestAPIResponceMessage)
        );
        msgValidationAPI.addData(
          getName(MessageEnum.RestAPIResponceDataMessage),
          msgValidationAPI.messageId
        );
        msgValidationAPI.addData(
          getName(MessageEnum.RestAPIResponceSuccessMessage),
          {
            data: []
          }
        );
        instance.commentApiCallId = msgValidationAPI.messageId;
        runEngine.sendMessage("Unit Test", msgValidationAPI);
  
        const msgError = new Message(
          getName(MessageEnum.RestAPIResponceErrorMessage)
        );
        msgError.addData(
          getName(MessageEnum.RestAPIResponceDataMessage),
          msgValidationAPI.messageId
        );
        msgError.addData(getName(MessageEnum.RestAPIResponceErrorMessage), {
          data: []
        });
        instance.commentApiCallId = msgValidationAPI.messageId;
        runEngine.sendMessage("Unit Test", msgValidationAPI);
      });

      then("I can leave the screen with out errors", () => {
        instance.componentWillUnmount();
        expect(commentsBlock).toBeTruthy();
      });
    });
  
});
