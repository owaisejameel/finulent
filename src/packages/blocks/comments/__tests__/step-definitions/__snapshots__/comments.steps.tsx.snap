// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comments User navigates to CreateComment 1`] = `
<ScrollViewMock
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "paddingHorizontal": 19,
      "paddingVertical": 16,
      "width": "100%",
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "height": 1134,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "marginBottom": 32,
            "marginVertical": 8,
            "textAlign": "left",
          }
        }
        testID="labelTitle"
      >
        Comments
      </Text>
      <View
        style={
          Object {
            "height": 1074,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onChangeText={[Function]}
          placeholder="Write a comment..."
          placeholderTextColor="#ddd"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "borderBottomWidth": 1,
                "color": "#252837",
                "fontSize": 16,
                "marginTop": 10,
                "paddingVertical": 10,
                "width": "100%",
              },
              Object {
                "borderBottomColor": "#eee",
              },
            ]
          }
          underlineColorAndroid="transparent"
        />
        <View
          style={
            Object {
              "bottom": 0,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <Button
            onPress={[Function]}
            title="Comment"
          />
        </View>
      </View>
    </View>
  </TouchableWithoutFeedback>
</ScrollViewMock>
`;
